pip/wheel.py

```diff
@@ -39,6 +44,57 @@
 logger = logging.getLogger(__name__)
 
 
+def _cache_for_filename(cache_dir, sdistfilename):
+    """Return a directory to store cached wheels in for sdistfilename.
+
+    Because there are M wheels for any one sdist, we provide a directory
+    to cache them in, and then consult that directory when looking up
+    cache hits.
+
+    We only insert things into the cache if they have plausible version
+    numbers, so that we don't contaminate the cache with things that were not
+    unique. E.g. ./package might have dozens of installs done for it and build
+    a version of 0.0...and if we built and cached a wheel, we'd end up using
+    the same wheel even if the source has been edited.
+
+    :param cache_dir: The cache_dir being used by pip.
+    :param sdistfilename: The filename of the sdist for which this will cache
+        wheels.
+    """
+    return os.path.join(cache_dir, 'wheels', sdistfilename)
+
+
+def cached_wheel(cache_dir, link):
+    if not cache_dir:
+        return link
+    if not link:
+        return link
+    if link.is_wheel:
+        return link
+    root = _cache_for_filename(cache_dir, link.filename)
+    try:
+        wheel_names = os.listdir(root)
+    except OSError as e:
+        if e.errno == errno.ENOENT:
+            return link
+        raise
+    candidates = []
+    for wheel_name in wheel_names:
+        try:
+            wheel = Wheel(wheel_name)
+        except InvalidWheelFilename:
+            continue
+        if not wheel.supported():
+            # Built for a different python/arch/etc
+            continue
+        candidates.append((wheel.support_index_min(), wheel_name))
```

Comment:
I was specifically looking for this sorting by the support index.  yay, it's here.